Missing #include <assert.h>

## Description

## Background Information / Reproduction Steps

Building v1.5.0 with GCC 15.2.0 fails for me with:

```
/build/source/tests/encryption/check_encryption_basic128rsa15.c: In function 'encryption_connect_fn':
/build/source/tests/encryption/check_encryption_basic128rsa15.c:219:5: error: implicit declaration of function 'assert' [-Wimplicit-function-declaration]
  219 |     assert(retval == UA_STATUSCODE_GOOD);
      |     ^~~~~~
/build/source/tests/encryption/check_encryption_basic128rsa15.c:27:1: note: 'assert' is defined in header '<assert.h>'; this is probably fixable by adding '#include <assert.h>'
   26 | #include "thread_wrapper.h"
  +++ |+#include <assert.h>
   27 |
```

Used CMake options:

```bash
-DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LOCALEDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/share/doc/open62541 -DCMAKE_INSTALL_INFODIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0/lib -DCMAKE_STRIP=/nix/store/a245z3cvf9x9sn0xlk6k8j9xhxbhda1z-gcc-wrapper-15.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/a245z3cvf9x9sn0xlk6k8j9xhxbhda1z-gcc-wrapper-15.2.0/bin/ranlib -DCMAKE_AR=/nix/store/a245z3cvf9x9sn0xlk6k8j9xhxbhda1z-gcc-wrapper-15.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/yjnvmby1w7giz1j9l1qc99yhrk9p9aaf-open62541-1.5.0 -DOPEN62541_VERSION:STRING=v1.5.0 -DUA_NAMESPACE_ZERO:STRING=FULL -DBUILD_SHARED_LIBS:BOOL=TRUE -DUA_BUILD_UNIT_TESTS:BOOL=TRUE -DUA_BUILD_EXAMPLES:BOOL=FALSE -DUA_ENABLE_ENCRYPTION:STRING=OPENSSL
```

## Checklist

Please provide the following information:

 - [x] open62541 Version (release number or git tag): v1.5.0
 - [ ] Other OPC UA SDKs used (client or server): 
 - [x] Operating system: Linux (NixOS)
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

--- a/tests/encryption/check_encryption_basic128rsa15.c
+++ b/tests/encryption/check_encryption_basic128rsa15.c
@@ -216,7 +216,6 @@ START_TEST(encryption_connect) {
 
     /* Secure client connect */
     retval = UA_Client_connect(client, "opc.tcp://localhost:4840");
-    assert(retval == UA_STATUSCODE_GOOD);
     ck_assert_uint_eq(retval, UA_STATUSCODE_GOOD);
 
     UA_Variant val;

